testsuite/tests/ui/oas/test_api_key_autocomplete.py

```diff
@@ -0,0 +1,80 @@
+"""Test for API key autocomplete in dev portal active docs"""
+import importlib_resources as resources
+import pytest
+
+from testsuite.ui.views.admin.audience.developer_portal import CMSEditPageView, \
+    CMSNewPageView
+from testsuite.ui.views.devel import DocsView
+from testsuite.utils import blame
+
+PAGE_CONTENT = """
+{% content_for javascripts %}
+    {{ 'active_docs.js' | javascript_include_tag }}
+{% endcontent_for %}
+
+{% assign spec = provider.api_specs['{name}']] %}
+
+<h1>Documentation</h1>
+
+<div class="swagger-section">
+  <div id="message-bar" class="swagger-ui-wrap">&nbsp;</div>
+  <div id="swagger-ui-container" class="swagger-ui-wrap"></div>
+</div>
+<script type="text/javascript">
+  (function () {
+    var url = "{{spec.url}}";
+    var serviceEndpoint = "{{spec.api_product_production_public_base_url}}"
+    SwaggerUI({ url: url, dom_id: "#swagger-ui-container" }, serviceEndpoint);
+  }());
+</script>
+"""
+
+
+@pytest.fixture(scope="module")
+def oas3_spec():
+    """OAS3 Active doc"""
+    return resources.files("testsuite.resources.oas3").joinpath("echo-api.json").read_text()
+
+
+# pylint: disable=unused-argument
+@pytest.fixture(scope="module")
+def api_doc_page(login, navigator, ui_active_doc, request, custom_admin_login):
+    """Custom doc page for developer portal"""
+    view = navigator.navigate(CMSNewPageView)
+    page_name = blame(request, "CustomDocumentation")
+    page_path = blame(request, "apidocs")
+
+    view.create(page_name, ". Root", page_path, PAGE_CONTENT.replace("{name}", ui_active_doc["system_name"]),
```

Comment:
`PAGE_CONTENT.format(name=ui_active_doc["system_name"])`

Reply:
It's not working for complex string with `{` and `}`